sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/SchemaValidation.scala

```diff
@@ -50,21 +51,49 @@ private[smartdatalake] trait SchemaValidation { this: DataObject =>
    * Validate the schema of a given Spark Data Frame `df` against `schemaMin`.
    *
    * @param df The data frame to validate.
+   * @param role role used in exception message. Set to read or write.
    * @throws SchemaViolationException is the `schemaMin` does not validate.
    */
-  def validateSchemaMin(df: DataFrame): Unit = {
-    schemaMin.foreach { structType =>
-      val diff = df.schemaDiffTo(structType,
+  def validateSchemaMin(df: DataFrame, role: String): Unit = {
+    schemaMin.foreach { schemaExpected =>
+      val missingCols = SchemaUtil.schemaDiff(schemaExpected, df.schema,
         ignoreNullable = Environment.schemaValidationIgnoresNullability,
         deep = Environment.schemaValidationDeepComarison
       )
-      if (diff.nonEmpty) {
+      if (missingCols.nonEmpty) {
         throw new SchemaViolationException(
-          s"""($id) does not have a valid schemaMin.
-             |- Actual schema: ${df.schema.treeString}
-             |- schemaMin: ${structType.treeString}
-             |- difference in: ${StructType(diff.toSeq).treeString}""".stripMargin)
+          s"""($id) DataFrame does not fulfil schemaMin on $role:
+             |- missingCols=${missingCols.mkString(", ")}
+             |- schemaMin: ${schemaExpected.fields.mkString(", ")}
+             |- schemaDf: ${df.schema.fields.mkString(", ")}""".stripMargin)
       }
     }
   }
+
+  /**
+   * Validate the schema of a given Spark Data Frame `df` against a given expected schema.
+   *
+   * @param df The data frame to validate.
+   * @param schemaExpected The expected schema to validate against.
+   * @param role role used in exception message. Set to read or write.
+   * @throws SchemaViolationException is the `schemaMin` does not validate.
+   */
+  def validateSchema(df: DataFrame, schemaExpected: StructType, role: String): Unit = {
```

Comment:
For the role, shouldn't we use an Enumeration here or do you think that's overkill?

Reply:
As we dont use it for any logic (no tests/matching on it), but it's just for the exception text i would leave it like this. Otherwise yes, should be an enumeration.